src/types/set.rs

```diff
@@ -71,96 +71,248 @@ impl PySet {
     where
         K: ToPyObject,
     {
-        fn inner(set: &PySet, key: PyObject) -> PyResult<bool> {
+        Bound::borrowed_from_gil_ref(&self).contains(key)
+    }
+
+    /// Removes the element from the set if it is present.
+    ///
+    /// Returns `true` if the element was present in the set.
+    pub fn discard<K>(&self, key: K) -> PyResult<bool>
+    where
+        K: ToPyObject,
+    {
+        Bound::borrowed_from_gil_ref(&self).discard(key)
+    }
+
+    /// Adds an element to the set.
+    pub fn add<K>(&self, key: K) -> PyResult<()>
+    where
+        K: ToPyObject,
+    {
+        Bound::borrowed_from_gil_ref(&self).add(key)
+    }
+
+    /// Removes and returns an arbitrary element from the set.
+    pub fn pop(&self) -> Option<PyObject> {
+        Bound::borrowed_from_gil_ref(&self).pop().map(|e| e.into())
```

Comment:
I'll use `Py::from` for now, I think later we might want a `.unbind()` method (`.into_unbound()`?) to go from `Bound` -> `Py`. Not sure what the best name for that is.

The return type here should probably have been `Option<&PyAny>` but I see little point in fixing the gil ref API now.

Reply:
> The return type here should probably have been Option<&PyAny> but I see little point in fixing the gil ref API now.

Very much agree.

Yes, for naming this, since our actual favourite and possible future name for `Py` was `PyDetached`, maybe `.detach()` would already be the right direction and does seem reasonably descriptive within the whole "bind to GIL" picture.

But I am would also be happy with `.unbind()` and I think both would be nicer than `Py::from` which is already nicer than `e.into()`.

Reply:
Pushed a commit which adds `Bound::unbind`, and maps over that here.